### PR TITLE
Update migration guide with registration API changes in 5.3.0

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: iOS-13-Migration-Guide.md [hunk 1]
```diff
@@ -65,7 +98,7 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
 3. Report the call to CallKit. Refer to this [example](https://github.com/twilio/voice-quickstart-swift/tree/master) for how to report the call to CallKit.
-4. Update how you decode the PushKit token. Previously we recommended providing the description field of the `PKPushCredentials.token`. Starting with iOS 13 the description format can change and become invalid preventing a user from registering for incoming call notifications.
+4. If you are using the Voice iOS SDK 5.2.1 and below, update how you decode the PushKit token. Previously we recommended providing the description field of the `PKPushCredentials.token`. Starting with iOS 13 the description format can change and become invalid preventing a user from registering for incoming call notifications.
```
Comment (aymenn) [inline iOS-13-Migration-Guide.md]: I think we should just talk about the latest release. We should remove this I think. Thoughts?

File: iOS-13-Migration-Guide.md [hunk 2]
```diff
@@ -97,7 +130,7 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+5. You must register via `[TwilioVoice registerWithAccessToken:deviceToken:completion:]` (Voice iOS SDK 5.2.1 and below) or `[TwilioVoice registerWithAccessToken:deviceTokenData:completion:]` (Voice iOS SDK 5.3.0 and above) when your App starts. This ensures that your app no longer receives “cancel” push notifications. A "call" push notification, when passed to `[TwilioVoice handleNotification:delegate:delegateQueue:]`, will return a `TVOCallInvite` object to you synchronously via the `[TVONotificationDelegate callInviteReceived:]` method when `[TwilioVoice handleNotification:delegate:delegateQueue:]` is called. A `TVOCancelledCallInvite` will be raised asynchronously via `[TVONotificationDelegate cancelledCallInviteReceived:error:]` if any of the following events occur:
```
Comment (aymenn) [inline iOS-13-Migration-Guide.md]: Same comment as above. Just the latest

File: iOS-13-Migration-Guide.md [hunk 3]
```diff
@@ -45,6 +45,39 @@ This section provides information required for existings apps built with Xcode 1
     Error Domain=com.twilio.voice.error Code=31301 "Http status: 400. Unexpected registration response." UserInfo={NSLocalizedDescription=Http status: 400. Unexpected registration response.}
     ```
 
+    If you are using the Voice iOS SDK 5.3.0 and above, pass the credential data as the argument directly to the `[TwilioVoice registerWithAccessToken:deviceTokenData:completion:]` method.
```
Comment (aymenn) [inline iOS-13-Migration-Guide.md]: As mentioned in another comment, we can just talk about the latest. Thoughts?

**Review by aymenn — CHANGES_REQUESTED** [review]

**Review by idelgado — APPROVED** [review]

File: iOS-13-Migration-Guide.md (hunk 3 above)
Comment (piyushtank) [reply iOS-13-Migration-Guide.md]: +1.  

Gave some thoughts after our discussion, the migration guide doesn't need to mention 2.x to <5.3.0 separately.  

The only drawback is if some customers have not followed the migration guide with 5.2.x SDK, they may want to see it - but in that case, our recommendation is to use the latest (5.3.0) sdk.

**Review by piyushtank — CHANGES_REQUESTED** [review]

File: iOS-13-Migration-Guide.md [hunk 4]
```diff
@@ -9,40 +9,37 @@ iOS 13 introduced changes in push notifications handling. This document describe
...
-    ```
-    
-    ```
-    Error Domain=com.twilio.voice.error Code=31301 "Http status: 400. Unexpected registration response." UserInfo={NSLocalizedDescription=Http status: 400. Unexpected registration response.}
+    ```.objective-c
+    - (void)pushRegistry:(PKPushRegistry *)registry 
+    didUpdatePushCredentials:(PKPushCredentials *)credentials 
+                 forType:(NSString *)type {
+        [TwilioVoice registerWithAccessToken:accessToken
+                             deviceTokenData:cachedDeviceToken
```
Comment (aymenn) [inline iOS-13-Migration-Guide.md]: Where’s `cachedDeviceToken` defined?

**Review by aymenn — CHANGES_REQUESTED** [review]

File: iOS-13-Migration-Guide.md (hunk 4 above)
Comment (bobiechen-twilio) [reply iOS-13-Migration-Guide.md]: Good catch. Fixed.

**Review by aymenn — APPROVED** [review]

**Pull request merged**